src/Tests/Microsoft.NET.TestFramework/EnvironmentInfo.cs

```diff
@@ -144,6 +144,18 @@ public static bool SupportsTargetFramework(string targetFramework)
             {
                 string restOfRid = currentRid.Substring(ridOS.Length + 1);
                 string osxVersionString = restOfRid.Split('-')[0];
+
+                //  Starting with macOS 12.0 (Monterey), the RID is "osx.12-x64" instead of the "osx.10.14-x64" format used before, so the version won't have a dot in it
+                if (!osxVersionString.Contains('.'))
```

Comment:
Instead of relying on the fact that the RID doesn't contain a period in it, could we make the OS parsing resilient to not having a dot?  Then this condition would be something like `macOSMajorVersion >= 12`.

Reply:
@dsplaisted good point, I switched to using float parsing for the version like we do for Ubuntu.